compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -253,7 +253,7 @@ impl MetaItem {
     ///
     /// In here, it'll return a span for `"foo"`.
     pub fn name_value_literal_span(&self) -> Option<Span> {
-        Some(self.name_value_literal()?.span)
+        self.name_value_literal().map(|lit| lit.span)
```

Comment:
Now this is becoming philosophical... or something :laughing: 

I personally really prefer the former code. I think we should use the `?` operator where reasonably usable instead of using `map` or `and_then`. This feels to me like using `async` and `.await` instead of `.map` and `.and_then` on futures.

Reply:
Heh, true, it is a bit philosophical. When reading the former code I didn't notice the `?` at first and wondered why this method signature has an `Option` when it only ever returns a `Some`. Decided to changed it to make that more obvious, but happy to change it back. Don't feel strongly either way.

And I have no intention of going and swapping `?` for `map` in general.

Reply:
Let's keep the current code then.

Reply:
Dropped.